### DISCUSSION TITLE
Unable to push large files even with LFS initialized

### DISCUSSION THREAD

**Discussion by digital-ember** [git-lfs/git-lfs#5194] [Q&A] [answered]
I might not be seeing the forest for the trees here. 

I just installed and initialized LFS for my repository. 
LFS tracks "*.jar" files:

![image](https://user-images.githubusercontent.com/19651078/203519327-6d83ce40-c82f-4825-9a62-0a73c11fa828.png)

`.gitattributes` was staged and committed.

When I try to push my commits, I still get this error:
![image](https://user-images.githubusercontent.com/19651078/203519490-15aa1a4f-b508-4064-8fdf-799fb9d35565.png)

When I add the files in question explicitely by file name ...
![image](https://user-images.githubusercontent.com/19651078/203521116-fbd740f2-ab54-434b-bc49-5c4d6262094d.png)

... my push still fails, but the size of object uploaded by LFS changes from 797MB to 859MB, so LFS seems to be handling these files, but the push to GitHub still fails.

Shouldn't the "*.jar" pattern work on the jar that causes the error? I wonder if it has to do with the filename containing a "dash"?

Can somebody please let me know what I'm doing wrong? Much appreciated!

**Comment by chrisd8088** [comment, marked as answer]
If you have a repository with existing files already committed, then just using `git lfs track` is not sufficient -- you will likely need to use `git lfs migrate import` to rewrite your repository's Git history so that the files are changed to Git LFS objects throughout, in every commit, both recent and historical.

There are some good [guides](https://confluence.atlassian.com/bitbucketserverkb/moving-git-large-files-to-git-lfs-in-bitbucket-server-1072486368.html) from Atlassian on options for this, and the full manual [page](https://github.com/git-lfs/git-lfs/blob/941b616aaf6f4adfa176616dc70c8d5e01b4a3d1/docs/man/git-lfs-migrate.adoc) offer further details on the `git lfs migrate` command, if you take that route.

In brief, `git lfs track` just creates a `.gitattributes` entry, which will in turn (given that you have Git LFS installed) only affect matching files in future commits.  If you already have large files in your Git history, you'll need to change that in some way.

**Reply by digital-ember** [reply]
Thanks Chris, perfect answer.
